lint/shellcheck.bzl

```diff
@@ -43,12 +43,19 @@ def shellcheck_action(ctx, executable, srcs, config, stdout, exit_code = None, o
     args.add_all(srcs)
     outputs = [stdout]
 
+    # Exit early when srcs is empty. This can happen when all srcs are generated files.
     if exit_code:
-        command = "{shellcheck} $@ >{stdout}; echo $? >" + exit_code.path
+        if len(srcs) == 0:
+            command = "touch {stdout} && echo 0 >" + exit_code.path
+        else:
+            command = "{shellcheck} $@ >{stdout} 2>&1; echo $? >" + exit_code.path
```

Comment:
what's the reasoning for sending both the stdout and stderr of shellcheck into the stdout file? I'd expect them to be meaningfully different, if they're not it suggests to me there's a bug in shellcheck. In that case I'd want to file it first before working around it.

Reply:
The reason why I added this was that shellcheck was reporting errors such ad the 'no files provided' to stderr, and the report was empty, while the exit code was non zero. It was hard to debug.

Reply:
yeah if it prints to stderr but exits zero then I agree that's hard to debug. I think if we wanted to print stderr somewhere it belongs under the `--@aspect_rules_lint//lint:debug` flag. We shouldn't mix the streams since stderr can contain stuff that doesn't belong in the report.

Reply:
What's the @aspect_rules_lint//lint:debug flag? How is that meant to be used?